pysparkling/tests/test_fromDDL.py

```diff
@@ -0,0 +1,121 @@
+"""
+Most of these test cases come from pyspark.sql.types._parse_datatype_string()
+"""
+import pytest
+
+from pysparkling.sql.types import (
+    ArrayType, ByteType, DateType, DecimalType, DoubleType, IntegerType, LongType, MapType, ShortType, StringType,
+    StructField, StructType
+)
+# DataType
+from pysparkling.sql.utils import ParseException
```

Comment:
StructType does not expose fromDDL in Python, (maybe we should not either), but the one in Scala does not show the behavior expected in these tests:

```scala
scala> StructType.fromDDL("some_str string, some_int integer, some_date date")
res2: org.apache.spark.sql.types.StructType = StructType(StructField(some_str,StringType,true), StructField(some_int,IntegerType,true), StructField(some_date,DateType,true))
```

```scala
scala> StructType.fromDDL("int")
org.apache.spark.sql.catalyst.parser.ParseException:
mismatched input '<EOF>'
```

(Spark 3)

Reply:
Multiple types of syntaxes are supported, fromDDL only handles one of them in Spark:
https://github.com/apache/spark/blob/master/python/pyspark/sql/types.py#L821-L841

Maybe we want not to rely on StructType.fromDDL to do this parsing?

Reply:
I think this method is wrongly called. We shouldn't call this fromDDL. This is actually `_parse_datatype_string`. And it's supported by Spark ;-). This is mainly used when you create a dataframe with a string as the field description.

Just check the implementation in pyspark, it supports multiple ways and I do agree that we should drop fromDDL (as it's not in pyspark at all).